vscode/src/client/analyzerClient.ts

```diff
@@ -117,19 +131,28 @@ export class AnalyzerClient {
     );
     this.rpcConnection.listen();
 
-    this.outputChannel.appendLine(`Started the kai rpc server, pid: ${this.kaiRpcServer.pid}`);
+    await Promise.race([
+      new Promise<void>((resolve) => {
+        kaiRpcServer.on("serverReportsReady", (pid) => {
+          this.outputChannel.appendLine(`*** kai rpc server [${pid}] reports ready`);
+          resolve();
+        });
+      }),
+      setTimeout(5000),
+    ]);
+
+    this.outputChannel.appendLine(`Started the kai rpc server, pid: [${pid}]`);
```

Comment:
Clever

Reply:
Old pattern that I've used before to cancel an operation that's running for too long.  Different use here, but still works well.